server/lib/tuist_web/controllers/api/bundles_controller.ex

```diff
@@ -20,6 +22,170 @@ defmodule TuistWeb.API.BundlesController do
 
   tags ["Bundles"]
 
+  operation :index,
+    summary: "List bundles for a project",
+    operation_id: "listBundles",
+    parameters: %{
+      account_handle: [
+        in: :path,
+        type: :string,
+        required: true,
+        description: "The handle of the account."
+      ],
+      project_handle: [
+        in: :path,
+        type: :string,
+        required: true,
+        description: "The handle of the project."
+      ],
+      git_branch: [
+        in: :query,
+        type: :string,
+        required: false,
+        description: "Filter bundles by git branch."
+      ],
+      page: [
+        in: :query,
+        type: :integer,
+        required: false,
+        description: "Page number for pagination."
+      ],
+      page_size: [
+        in: :query,
+        type: :integer,
+        required: false,
+        description: "Number of items per page."
+      ]
+    },
+    responses: %{
+      ok:
+        {"List of bundles", "application/json",
+         %Schema{
+           type: :object,
+           properties: %{
+             data: %Schema{
+               type: :array,
+               items: BundleList
+             },
+             meta: %Schema{
+               type: :object,
+               properties: %{
+                 current_page: %Schema{type: :integer},
+                 page_size: %Schema{type: :integer},
+                 total_count: %Schema{type: :integer},
+                 total_pages: %Schema{type: :integer}
+               },
+               required: [:current_page, :page_size, :total_count, :total_pages]
+             }
+           },
+           required: [:data, :meta]
+         }},
+      unauthorized: {"You need to be authenticated to list bundles", "application/json", Error},
+      forbidden: {"You are not authorized to list bundles", "application/json", Error}
+    }
+
+  def index(%{assigns: %{selected_project: selected_project}} = conn, params) do
+    # Create filters in the format Flop expects (list of filter objects)
+    filters = [
+      %{field: :project_id, op: :==, value: selected_project.id}
+    ]
+
+    filters =
+      case Map.get(params, "git_branch") do
+        nil -> filters
+        branch -> [%{field: :git_branch, op: :==, value: branch} | filters]
+      end
+
+    flop_params = %{
+      filters: filters,
+      order_by: [:inserted_at],
+      order_directions: [:desc],
+      page_size: Map.get(params, "page_size", 50)
+    }
+
+    flop_params =
+      case Map.get(params, "page") do
+        nil -> flop_params
+        page -> Map.put(flop_params, :page, page)
+      end
+
+    case Bundles.list_bundles(flop_params, preload: [:uploaded_by_account, project: :account]) do
+      {bundles, meta} ->
+        conn
+        |> put_status(:ok)
+        |> json(%{
+          data: Enum.map(bundles, &bundle_list_to_map/1),
```

Comment:
maybe would call this `items` instead of `data`, but don't feel too strongly

Reply:
I renamed it to `bundles` to align with [this](https://tuist.dev/api/docs#tag/runs/get/api/projects/{account_handle}/{project_handle}/runs), which uses `runs`.